lib/spacesuit/cors_middleware.ex

```diff
@@ -139,7 +139,11 @@ defmodule Spacesuit.CorsMiddleware do
   end
 
   defp valid_control_headers?(headers) do
-    MapSet.subset?(headers, access_control_request_headers())
+    if MapSet.size(headers) == 0 do
```

Comment:
The body of this whole method can just read:

```
MapSet.size(headers) == 0 || MapSet.subset?(headers, access_control_request_headers())
```

You can get rid of the whole `if`/`else` block